src/storage/heap_file.c

```diff
@@ -20657,12 +20657,14 @@ heap_get_insert_location_with_lock (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONT
 	{
 	  /* successfully locked! */
 #if defined (SERVER_MODE)
-	  if (lock == SCH_M_LOCK && OID_IS_ROOTOID (&context->class_oid))
+	  const bool is_lock_for_class = OID_IS_ROOTOID (&context->class_oid) && lock == SCH_M_LOCK;
```

Comment:
I think we need a descriptive comment here. It was only for class, and the code itself describes it. However, now we take the serial class into account and this code doesn't describe why. 
```suggestion
          // leaving a lock information for DDL. The insertion to the db_serial means a new serial is created with the DDL like create serial ..
	  const bool is_lock_for_class = OID_IS_ROOTOID (&context->class_oid) && lock == SCH_M_LOCK;
```

Reply:
The same goes for others leaving the log record as well.

Reply:
OK, I will leave a comment